src/__tests__/downshift.get-input-props.js

```diff
@@ -90,55 +90,177 @@ test('manages arrow up and down behavior', () => {
+  })
 
-test('arrow down opens menu and highlights defaultHighlightedIndex + 1 if provided', () => {
-  const defaultHighlightedIndex = 2
-  const {arrowDownInput, childrenSpy} = renderDownshift({
-    props: {defaultHighlightedIndex},
+  test('defaultHighlightedIndex + 1', () => {
+    const defaultHighlightedIndex = 2
+    const {arrowDownInput, childrenSpy} = renderDownshift({
+      // provide only defaultHighlightedIndex
+      props: {defaultHighlightedIndex},
+    })
+    // ↓
+    arrowDownInput()
+    expect(childrenSpy).toHaveBeenLastCalledWith(
+      expect.objectContaining({
+        isOpen: true,
+        highlightedIndex: defaultHighlightedIndex + 1,
+      }),
+    )
+  })
+
+  test('initialHighlightedIndex + 1 then defaultHighlightedIndex + 1', () => {
+    const initialHighlightedIndex = 3
+    const defaultHighlightedIndex = 2
+    const {arrowDownInput, escapeOnInput, childrenSpy} = renderDownshift({
+      // provide both initialHighlightedIndex defaultHighlightedIndex
+      props: {defaultHighlightedIndex, initialHighlightedIndex},
+    })
+    // ↓
+    arrowDownInput()
+    expect(childrenSpy).toHaveBeenLastCalledWith(
+      expect.objectContaining({
+        isOpen: true,
+        highlightedIndex: initialHighlightedIndex + 1,
+      }),
+    )
+    escapeOnInput()
+    // ↓
+    arrowDownInput()
+    expect(childrenSpy).toHaveBeenLastCalledWith(
+      expect.objectContaining({
+        isOpen: true,
+        highlightedIndex: defaultHighlightedIndex + 1,
+      }),
+    )
+  })
+
+  test('highlightedIndex if controlled', () => {
+    const highlightedIndex = 2
+    const {arrowDownInput, childrenSpy} = renderDownshift({
+      // control highlightedIndex
+      props: {highlightedIndex},
+    })
+    // ↓
+    arrowDownInput()
+    expect(childrenSpy).toHaveBeenLastCalledWith(
+      expect.objectContaining({
+        isOpen: true,
+        highlightedIndex,
+      }),
+    )
   })
-  // ↓
-  arrowDownInput()
-  expect(childrenSpy).toHaveBeenLastCalledWith(
-    expect.objectContaining({
-      isOpen: true,
-      highlightedIndex: defaultHighlightedIndex + 1,
-    }),
-  )
 })
 
-test('arrow up opens menu and highlights defaultHighlightedIndex - 1 if provided', () => {
-  const defaultHighlightedIndex = 2
-  const {arrowUpInput, childrenSpy} = renderDownshift({
-    props: {defaultHighlightedIndex},
+describe('arrow up opens menu and highlights item at index', () => {
+  test('length - 1 by default', () => {
+    const {arrowUpInput, childrenSpy} = renderDownshift()
+    // ↑
+    arrowUpInput()
+    expect(childrenSpy).toHaveBeenLastCalledWith(
+      expect.objectContaining({
+        isOpen: true,
+        highlightedIndex: colors.length - 1,
+      }),
+    )
+  })
+
+  test('initialHighlightedIndex - 1', () => {
+    const initialHighlightedIndex = 3
+    const {arrowUpInput, childrenSpy} = renderDownshift({
+      // provide only highlightedIndex
+      props: {initialHighlightedIndex},
+    })
+    // ↓
```

Comment:
should be `// ↑` in this and the following

Reply:
thanks, will change!